registry/storage/cache/memory/memory.go

```diff
@@ -30,39 +50,59 @@ func (imbdcp *inMemoryBlobDescriptorCacheProvider) RepositoryScoped(repo string)
 		return nil, err
 	}
 
-	imbdcp.mu.RLock()
-	defer imbdcp.mu.RUnlock()
-
 	return &repositoryScopedInMemoryBlobDescriptorCache{
-		repo:       repo,
-		parent:     imbdcp,
-		repository: imbdcp.repositories[repo],
+		repo:   repo,
+		parent: imbdcp,
 	}, nil
 }
 
 func (imbdcp *inMemoryBlobDescriptorCacheProvider) Stat(ctx context.Context, dgst digest.Digest) (distribution.Descriptor, error) {
-	return imbdcp.global.Stat(ctx, dgst)
+	if err := dgst.Validate(); err != nil {
+		return distribution.Descriptor{}, err
+	}
+
+	key := descriptorCacheKey{
+		digest: dgst,
+	}
+	descriptor, ok := imbdcp.lru.Get(key)
+	if ok {
+		return descriptor.(distribution.Descriptor), nil
```

Comment:
I _suspect_ skipping type assertion here will trigger OSS fuzzer (event though it does not make much sense here)

Reply:
I've added type assertions just in case.